### PR TITLE
Zadanie 3

### PR DESCRIPTION
**Pull request by wojt157**

### CODE REVIEW

**Review by KotwinskiDiego — COMMENTED** [review]
Czesc,
Wszystko zgodnie z trescia zadania, jednak zabraklo mi próby wytrenowania lepszego modelu.
Mogłeś:
- sprobowac odpalic trening dla wiekszej ilosci epok
- pobawic sie base czy innymi hiperparametrami
- sprawdzić inne sposoby augmentacji
- jest pewnie wiele innych sposobow o ktorych ja tez nie wiem

**Review by alicjafinger — COMMENTED** [review]
val_pixel_acc rosnie, a train_lloss maleje (oprocz ostatniej epoki) i jest wiele epok, wg mnie zgodnie z wymaganiami wszystko zrobione git

**Review by KLTPL — APPROVED** [review]
Warunki zadania jak najbardziej spełnione 👍️
Możnaby wyświetlić więcej wyników, żeby zobaczyć dokładniej jak model działa i dodać trochę epok, żeby zwiększyć celność i żeby coś więcej rysował.
Poza tym super

**Comment by jgilewicz** [comment]
Zadanie wykonane bardzo dobrze. Całość jest poprawna. Zadanie zaliczone. Kilka drobnych uwag na przyszłość: 
- `nn.Conv2d(in_ch, out_ch, kernel_size=3, padding=1)` staraj się nie zostawiać magic number w kodzie. Lepiej ustawić to defaultowo w konstruktorze.
- jeszcze komentarz odnośnie samych wyników (czysto informacyjny, bo zadanie trochę źle zostało sformułowane) - tak mały UNet nie poradzi sobie z segmentacją, potrzebne jest znaczące zwiększenie modelu + wydłużenie treningu na większym modelu. Tutaj podsyłam do wglądu w [notatniku](https://github.com/keyurparalkar/Semantic-Segmentation-with-UNETs/blob/master/FASTai_UNET.ipynb) jak duży jest ten UNet zaimplementowany za pomocą [FastAI](https://github.com/fastai/fastai).
